djweb/dj_comp_hist/views.py

```diff
@@ -187,4 +187,69 @@ def advanced_search(request):
     :param request:
     :return:
     """
-    return HttpResponse("work in progress")
+    boxes = []
+    if "checkBox1" in request.GET:
+        boxes.append(1)
+    if "checkBox2" in request.GET:
+        boxes.append(2)
+    if "checkBox3" in request.GET:
+        boxes.append(3)
+    if len(boxes) == 2:
+        doc_objs = Document.objects.filter(Q(folder__box__number=boxes[0]) |
+                                           Q(folder__box__number=boxes[1]))
+    elif len(boxes) == 1:
+        doc_objs = Document.objects.filter(folder__box__number=boxes[0])
+    else:
+        doc_objs = Document.objects
+
+    try:
+        author = request.GET['author']
+        if author != "Unknown":
+            author = author.split(" ")
+            doc_objs = doc_objs.filter(Q(author_person__first__icontains=author[0]) |
+                                       Q(author_person__last__icontains=author[0]) |
+                                       Q(author_organization__name__icontains=author[0]))
+    except:
+        print("Error getting author name")
+
+    try:
+        recipient = request.GET['receiver']
+        if recipient != "Unknown":
+            recipient = recipient.split(" ")
+            doc_objs = doc_objs.filter(Q(recipient_person__first__icontains=recipient[0]) |
+                                       Q(recipient_person__last__icontains=recipient[0]) |
+                                       Q(recipient_organization__name__icontains=recipient[0]))
+    except:
+        print('Error getting recipient name')
+
+    try:
```

Comment:
From manually testing the search function, I found that if you select more than one document type, it only gives you results for the last one that you selected.

Reply:
finally completed